packages/openapi/src/decorators.ts

```diff
@@ -0,0 +1,65 @@
+import { Program, Type } from "@cadl-lang/compiler";
+import { reportDiagnostic } from "./lib.js";
+
+const operationIdsKey = Symbol();
+export function $operationId(program: Program, entity: Type, opId: string) {
+  if (entity.kind !== "Operation") {
+    reportDiagnostic(program, {
+      code: "decorator-wrong-type",
+      format: { decorator: "operationId", entityKind: entity.kind },
+      target: entity,
+    });
+    return;
+  }
+  program.stateMap(operationIdsKey).set(entity, opId);
+}
+
+export function getOperationId(program: Program, entity: Type): string | undefined {
+  return program.stateMap(operationIdsKey).get(entity);
+}
+
+const pageableOperationsKey = Symbol();
+export function $pageable(program: Program, entity: Type, nextLinkName: string = "nextLink") {
```

Comment:
Should that decorator be part of `openapi`? It is not used by the openapi3 emitter. There is a few specs in the public repo using it but maybe they shouldn't

Reply:
This uses `x-ms-pageable` in openAPI, but in openapi3, this would be represented as an operation link

Reply:
I seem to recall long time back we weren't super sure where to put pageable - probably cadl core needs some notion of pageable given it's a common pattern, but we opted for putting it in a leaf package for now.

Reply:
Yeah but in its current state it only takes in the name of the property in the body referring to the next link. I feel this is a little too `cadl-autorest` specific and I should remove this decorator from teh shared openapi library and leave it in cadl-autorest until we figure out a good paging story for core

Reply:
ok so did find an issue with not having it here. In the cadl-azure repo we build for both autorest and openapi target everything and if they use `@pageable` in the spec then it isn't available when targeting openapi3. I think I will then put back pageable in openapi3 for now and remove it after the emitter api makes it in as it should solve that problem